### ISSUE TITLE
Abandon LogDomains loggers and use System.Logger

### ISSUE THREAD

**Issue by dmatej** [eclipse-ee4j/glassfish#24173]
### Why

1. It is really hard to find something when half of logger names is obfuscated by LogDomains
2. Prefer System.Logger (with GJULE as a backend, but could use any logging framework with some limitations)
3. Current state is a growing mess (take a look on logging.properties).
4. System.Logger is already preferred.
5. Usages are more readable. Some usages are pretty ineffective, they deserve review. 

### How

It will be a lot of work, but quite easy. Most of changes could be probably done by Eclipse regexp finder.

1. First do https://github.com/eclipse-ee4j/glassfish/issues/23627, don't start with this without that.
2. Relates to #24064 and should be done before that.
3. Bind logger names to the class name (standard)
4. Simplify usages (no array construction required)

**Comment by renatsaf** [comment]
I'd like to pick this up. I measured what is actually left on `main` (c8ac1be) and have a few questions before writing any code, because three decisions here are one-way and I'd rather not make them unilaterally across 250 files.

### What is left

`com.sun.logging.LogDomains` + `LogDomainsLogger` in `nucleus/common/common-util` are still referenced by **275 `.java` files** (250 under `src/main`, 18 under `src/test`, the rest in the non-built `appserv-tests` / `v2-tests` trees), for **281 `LogDomains.getLogger` call sites**. It is very concentrated — five modules are ~60% of it:

| module | files | main constant |
|---|---|---|
| `appserver/connectors/connectors-runtime` | 63 | `RSR_LOGGER` (114 uses overall) |
| `appserver/transaction/jts` | 45 | `TRANSACTION_LOGGER` (46) |
| `appserver/ejb/ejb-container` | 24 | `EJB_LOGGER` (31) |
| `appserver/jdbc/jdbc-ra` | 23 | `RSR_LOGGER` |
| `appserver/security/core-ee` | 12 | `SECURITY_LOGGER` (19) |

The rest is a tail of ~20 modules with 1–7 files each.

The infrastructure side is already done — `GlassFishLoggerFinder` is in place, 333 files already use `System.getLogger(...)`, and there is an established pattern to copy from 35dc4b3843 (appclient.security), 42ed92cdfb (WebappClassLoader) and 4f8cf57580 (resource pool). So this is finishing work, not design work.

### Questions

**1. Is log-message i18n intentionally being dropped?**

There are 40 `LogStrings.properties` files with ~1899 keys, and ~350 call sites pass a bundle key as the message. `System.Logger` resolves no bundle on `log(Level, String, Object...)`; the bundle-aware overloads need the `ResourceBundle` passed at every call. Both migration commits above resolve this by inlining the English text and deleting the key from `LogStrings.properties`, which I read as "localized log messages are going away". I'm happy to follow that, I'd just like it on the record before I delete a few hundred keys — and note it ratchets: a shared bundle file can't be removed until the last class under it is migrated (`connectors-runtime` has 283 keys, `war-util` 252, `ejb-container` 203).

**2. How should `CONFIG` and `FINEST` map?**

`System.Logger` has neither, and no `logp`/`entering`/`exiting`. Across the 250 files: `FINE` 1238, `WARNING` 314, `SEVERE` 243, `FINEST` 156, `INFO` 114, `FINER` 62, `CONFIG` 25, plus 108 `logp(...)` and 597 `isLoggable(...)` guards. `SEVERE→ERROR`, `FINE→DEBUG`, `FINER→TRACE` are unambiguous. `FINEST→TRACE` collapses two levels, and `CONFIG→?` has no obvious answer. Since GJULE maps `TRACE→FINER` / `DEBUG→FINE`, anything currently emitted only at `FINEST` or `CONFIG` changes its effective threshold for users. One convention decided here beats 250 ad-hoc ones.

**3. Can the #23627 ordering be relaxed?**

The description says to do #23627 first. That's an umbrella with one starter PR so far, so taken strictly it blocks this indefinitely. The concern behind it is real — binding logger names to class names and then renaming packages changes user-visible logger names twice. Would you accept "where a module is in scope for #23627, do the package rename and the logger migration in the same PR; elsewhere proceed now"?

### Two things that need handling either way

- **Config compatibility.** `LogDomains` names a logger `prefix + "." + package`, e.g. `jakarta.enterprise.resource.resourceadapter.com.sun.enterprise.connectors`. Class-named loggers remove that `jakarta.*` subtree, so the **96 `jakarta.*` entries** in `nucleus/admin/template/src/main/resources/config/logging.properties` go dead and any user `set-log-levels jakarta.enterprise.…` stops taking effect. The same file already has package-named entries (`com.sun.enterprise.connectors.level`, …) that cover the new names, so the fix is pruning the template plus an upgrade note and edits to `docs/administration-guide/.../logging.adoc`, `jdbc.adoc` and `general-administration.adoc`.
- **`LogDomains` is public shipped API** in `common-util`. I'd suggest `@Deprecated(forRemoval = true)` for one release before deleting it, rather than removing it outright.

Also worth a decision, though possibly out of scope here: `list-loggers` / `list-log-levels` / the admin console log-level page are fed by `LoggerInfoMetadata` via `@LoggerInfo` (a separate registry, ~226 files). Class-named loggers won't show up there. Is `@LoggerInfo` also on its way out, or should migrated loggers be registered?

### Proposed sequencing

Module-by-module, ~6–7 self-contained PRs rather than one huge one or thirty small ones, to keep CI load reasonable:

1. small leaf modules (`gms-adapter`, `resources/mail`, `container-common`, `orb-connector`, `jpa-container`, `appclient/client` — ~27 files) to lock the conventions from the answers above;
2. `jdbc-ra` + `jdbc-runtime` (28) — self-contained bundles under `com/sun/gjc/**`, so `LogStrings.properties` can actually be emptied;
3. `ejb-container` + `ejb-full-container` + `entitybean-container` (34);
4. `transaction/jts` + `jta` (51);
5. `connectors-runtime` + `connectors-internal-api` + inbound (73, partly pre-migrated by 4f8cf57580);
6. `security/core-ee` + `ejb.security` + tail;
7. closing PR: deprecate/remove `LogDomains` + `LogDomainsLogger`, prune the dead `jakarta.*` lines from the domain template, update the three admin-guide adocs, add the upgrade note.

If that split works for you and you can answer (1)–(3), I'll start with step 1.
